packages/compass-preferences-model/src/preferences.ts

```diff
@@ -89,25 +89,53 @@ export type AmpersandType<T> = T extends string
   : never;
 
 type PreferenceDefinition<K extends keyof GlobalPreferences> = {
+  /** The type of the preference value, in Ampersand naming */
   type: AmpersandType<GlobalPreferences[K]>;
+  /** An optional default value for the preference */
   default?: GlobalPreferences[K];
+  /** Whether the preference is required in the Ampersand model */
   required: boolean;
+  /** Whether the preference can be modified through the Settings UI */
   ui: K extends keyof UserConfigurablePreferences ? true : false;
+  /** Whether the preference can be set on the command line */
   cli: K extends keyof InternalUserPreferences
     ? false
     : K extends keyof CliOnlyPreferences
     ? true
     : boolean;
+  /** Whether the preference can be set in the global config file */
   global: K extends keyof InternalUserPreferences
     ? false
     : K extends keyof CliOnlyPreferences
     ? false
     : boolean;
+  /** A description used for the --help text and the Settings UI */
   description: K extends keyof InternalUserPreferences
     ? null
     : { short: string; long?: string };
+  /** A method for deriving the current semantic value of this option, even if it differs from the stored value */
+  deriveValue?: DeriveValueFunction<GlobalPreferences[K]>;
```

Comment:
If we are allowing to have derived values, should we also account for this in the `onPreferenceValueChanged` method? It seems like right now I can be subscribed to changes of e.g., `trackErrors` value without getting an update if `networkTraffic` was changed

Reply:
Yes, great point! Done in 8a2e563cc71845710af445f7863ce06bc499f08e